.github/workflows/saffron.yml

```diff
@@ -0,0 +1,40 @@
+name: Saffron CI
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  run:
+    name: Run saffron e2e tests
+
+    runs-on: ["ubuntu-latest"]
+
+    strategy:
+      matrix:
+        rust_toolchain_version: ["1.74"]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Use shared Rust toolchain setting up steps
+        uses: ./.github/actions/toolchain-shared
+        with:
+          rust_toolchain_version: ${{ matrix.rust_toolchain_version }}
+
+      - name: Apply the Rust smart cacheing
+        uses: Swatinem/rust-cache@v2
+
+      - name: Build the saffron cli binary
+        run: |
+          cargo build --release --bin saffron
+
+      - name: Run the saffron e2e encoding tests
+        run: |
+          ./saffron/test-encoding.sh saffron/fixtures/lorem.txt
```

Comment:
You're building in release mode above, but the script doesn't use the release flag. Therefore, the command above is useless.

Reply:
:man_facepalming: